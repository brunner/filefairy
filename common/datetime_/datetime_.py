#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import datetime
import pytz
import re

_est = pytz.timezone('America/New_York')
_pst = pytz.timezone('America/Los_Angeles')


def datetime_as_pst(d):
    return d.astimezone(_pst)


def datetime_datetime_est(*args):
    return _est.localize(datetime.datetime(*args))


def datetime_datetime_pst(*args):
    return _pst.localize(datetime.datetime(*args))


def datetime_now():
    return _pst.localize(datetime.datetime.now())


def decode_datetime(s):
    return datetime_datetime_pst(*map(int, re.findall('\d+', s[:-6])))


def encode_datetime(d):
    return datetime_as_pst(d).isoformat()


def suffix(day):
    return 'th' if 11 <= day <= 13 else {
        1: 'st',
        2: 'nd',
        3: 'rd'
    }.get(day % 10, 'th')


def timedelta(then, now):
    diff = now - then
    s, d = diff.seconds, diff.days

    if s < 0 or d < 0:
        return timedelta(now, then)

    if d > 0:
        s += 86400 * d

    h, m = s // 3600, (s // 60) % 60
    hm = []
    if h > 0:
        hm.append('{}h'.format(h))
    hm.append('{}m'.format(m))

    return ' '.join(hm)


def timestamp(date):
    return date.strftime('%H:%M:%S %Z (%Y-%m-%d)')
