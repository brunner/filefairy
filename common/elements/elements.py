#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""Common (non-reloadable) util methods for rendering templates."""

FAVICON_LINK = 'https://fairylab.surge.sh/favicon-32x32.png'


def cell(col=None, content=''):
    """Builds a cell object, describing a table cell.

    Args:
        col: The col object for this particular cell.
        content: The cell text.

    Returns:
        The cell object.
    """
    obj = {}
    if col:
        obj['col'] = col
    if content:
        obj['content'] = content
    return obj


def col(clazz='', colspan=''):
    """Builds a col object, describing a table column.

    Args:
        clazz: The CSS class(es) to apply to the column.
        colspan: The number of columns that this column should span.

    Returns:
        The col object.
    """
    obj = {}
    if clazz:
        obj['clazz'] = clazz
    if colspan:
        obj['colspan'] = colspan
    return obj


SITELINKS_HCOLS = [col(clazz='font-weight-bold text-dark')]
SITELINKS_BCOLS = [col(clazz='position-relative')]
SITELINKS_TABLES = [
    ('Tasks', [
        ('timer', '/gameday/', 'Gameday'),
        ('people', '/news/', 'News'),
        ('spreadsheet', '/standings/', 'Standings'),
    ]),
    ('Other', [
        ('dashboard', '/dashboard/', 'Dashboard'),
        ('home', '/', 'Home'),
    ]),
]


def anchor(url, content):
    """Builds an anchor element.

    Args:
        url: The link url.
        content: The inner HTML of the anchor element.

    Returns:
        The anchor element.
    """
    return '<a href="{}">{}</a>'.format(url, content)


def dialog(id_='', icon='', tables=None):
    """Builds a dialog object.

    Args:
        id_: The CSS id to set on the root dialog element.
        icon: The icon element to display in the dialog element header.
        tables: The optional list of table elements.

    Returns:
        The dialog object.
    """
    obj = {}
    if id_:
        obj['id'] = id_
    if icon:
        obj['icon'] = icon
    if tables:
        obj['tables'] = tables
    return obj


def icon_img(src, size, classes):
    """Builds an icon element for an image resource.

    Args:
        src: The source of the image.
        size: The size of the image, in pixels.
        classes: The list of CSS classes to apply to the img.

    Returns:
        The span element.
    """
    classes = ' class="{}"'.format(' '.join(classes)) if classes else ''
    tag = '<img src="{0}" width="{1}" height="{1}" border="0"{2}>'
    return tag.format(src, size, classes)


def icon_span(name='', attributes=None, classes=None):
    """Builds an icon element for an Iconic font span.

    Args:
        name: The name of the icon.
        attributes: The set of data attributes to apply to the span.
        classes: The list of CSS classes to apply to the span.

    Returns:
        The span element.
    """
    if classes is None:
        classes = []
    classes = ['oi', 'oi-' + name, 'absolute-icon'] + classes
    return span(classes=classes, attributes=attributes)


def menu():
    """Builds a menu dialog.

    Returns:
        The menu dialog.
    """
    img = icon_img(FAVICON_LINK, '16', ['absolute-icon', 'left'])
    span_ = span(classes=['d-block', 'px-4'], text='Fairylab')
    return dialog(id_='menu', icon=(img + span_), tables=sitelinks(home=True))


def pre(content):
    """Builds a pre element.

    Args:
        content: The inner HTML of the pre element.

    Returns:
        The pre element.
    """
    return '<pre>{}</pre>'.format(content)


def ruleset(selector='', rules=None):
    """Builds a CSS ruleset element.

    Args:
        selector: The CSS selector.
        rules. The list of rules.

    Returns:
        The ruleset element.
    """
    obj = {}
    if selector:
        obj['selector'] = selector
    if rules:
        obj['rules'] = rules
    return obj


def sitelinks(home=False):
    """Builds a list of sitelinks tables.

    Args:
        home: Whether or not to include a Home link in the sitelinks.

    Returns:
        The list of sitelinks tables.
    """
    tables = [topper('Site Links')]
    for head, rows in SITELINKS_TABLES:
        body = []
        for name, href, text in rows:
            if name == 'home' and not home:
                continue
            icon = icon_span(name=name, classes=['left', 'text-secondary'])
            span_ = span(classes=['d-block pl-4'], text=anchor(href, text))
            body.append([cell(content=(icon + span_))])

        tables.append(
            table(
                clazz='border mb-3',
                hcols=SITELINKS_HCOLS,
                bcols=SITELINKS_BCOLS,
                head=[[cell(content=head)]],
                body=body))

    return tables


def span(classes=None, attributes=None, text=''):
    """Builds a span element.

    Args:
        classes: The list of CSS classes to apply to the span.
        attributes: The set of data attributes to apply to the span.
        content: The inner HTML of the span element.

    Returns:
        The span element.
    """
    if attributes is None:
        attr = ''
    else:
        items = sorted(attributes.items())
        attr = ' ' + ' '.join('{}="{}"'.format(k, v) for k, v in items)
    if classes is None:
        classes = []
    return '<span class="{}"{}>{}</span>'.format(' '.join(classes), attr, text)


def table(clazz='',
          id_='',
          hcols=None,
          bcols=None,
          fcols=None,
          head=None,
          body=None,
          foot=None):
    """Builds a table element.

    Args:
        clazz: The optional CSS class to set on the root table element.
        id_: The optional CSS id to set on the root table element.
        hcols: The optional list of CSS classes to apply to each head column.
        bcols: The optional list of CSS classes to apply to each body column.
        fcols: The optional list of CSS classes to apply to each footer column.
        head: The optional list of head rows. Each row is a list of cells.
        body: The optional list of body rows. Each row is a list of cells.
        foot: The optional list of footer rows. Each row is a list of cells.

    Returns:
        The table element.
    """
    obj = {}
    if clazz:
        obj['clazz'] = clazz
    if id_:
        obj['id'] = id_
    if hcols:
        obj['hcols'] = hcols
    if bcols:
        obj['bcols'] = bcols
    if fcols:
        obj['fcols'] = fcols
    if head:
        obj['head'] = head
    if body:
        obj['body'] = body
    if foot:
        obj['foot'] = foot
    return obj


def topper(text):
    """Builds a topper table element.

    Args:
        text: The text of the topper table element.

    Returns:
        The topper table element.
    """
    return table(
        clazz='topper',
        bcols=[col(clazz='border-0 font-weight-bold px-0 text-secondary')],
        body=[[cell(content=text)]])
