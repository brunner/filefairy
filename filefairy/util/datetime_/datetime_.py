#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import datetime
import re


def decode_datetime(s):
    return datetime.datetime(*map(int, re.findall('\d+', s)))


def encode_datetime(d):
    return d.isoformat()


def suffix(day):
    return 'th' if 11 <= day <= 13 else {
        1: 'st',
        2: 'nd',
        3: 'rd'
    }.get(day % 10, 'th')
