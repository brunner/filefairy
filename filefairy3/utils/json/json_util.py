#!/usr/bin/env python

import json


def dumps(data):
    return json.dumps(data, indent=2, sort_keys=True)
