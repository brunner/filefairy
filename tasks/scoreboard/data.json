{
  "finished": false,
  "highlights": {},
  "injuries": {},
  "offseason": false,
  "postseason": true,
  "scores": {
    "2024-11-04T00:00:00-08:00": [
      "<{0}{1}25236.html|T32 2, T47 1>"
    ]
  },
  "started": false,
  "table": {},
  "unchecked": []
}
