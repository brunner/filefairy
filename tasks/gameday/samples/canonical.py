#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""Sample data for gameday.py golden test."""

import json
import os
import re
import sys

_path = os.path.dirname(os.path.abspath(__file__))
sys.path.append(re.sub(r'/tasks/gameday/samples', '', _path))

from common.datetime_.datetime_ import datetime_datetime_pst  # noqa
from common.datetime_.datetime_ import encode_datetime  # noqa
from common.elements.elements import topper  # noqa
from common.test.test import get_testdata  # noqa
from services.scoreboard.scoreboard import line_score_hide_body  # noqa
from services.scoreboard.scoreboard import line_score_hide_foot  # noqa
from services.scoreboard.scoreboard import pending_hide_body  # noqa

DATE_08280000 = datetime_datetime_pst(2024, 8, 28)

TESTDATA = get_testdata()

_game_2449 = json.loads(TESTDATA['2449.json'])
_head_2449 = topper('Wednesday, August 28th, 2024')
_body_2449 = line_score_hide_body(_game_2449)
_foot_2449 = line_score_hide_foot(_game_2449)

_game_2469 = json.loads(TESTDATA['2469.json'])
_head_2469 = topper('Thursday, August 29th, 2024')
_body_2469 = line_score_hide_body(_game_2469)
_foot_2469 = line_score_hide_foot(_game_2469)

_game_2476 = json.loads(TESTDATA['2476.json'])
_head_2476 = topper('Friday, August 30th, 2024')
_body_2476 = line_score_hide_body(_game_2476)
_foot_2476 = line_score_hide_foot(_game_2476)

_body_pending = pending_hide_body(
    encode_datetime(DATE_08280000),
    ['T31 4, TLA 2', 'TNY 5, TLA 3', 'TNY 1, T55 0'])

_days = [
    [_head_2449, _body_2449, _foot_2449, _body_pending],
    [_head_2469, _body_2469, _foot_2469],
    [_head_2476, _body_2476, _foot_2476],
]

subtitle = ''

tmpl = 'gameday.html'

context = {
    'days': _days,
}
