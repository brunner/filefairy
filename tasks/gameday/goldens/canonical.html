<!doctype html>
<html lang="en">

<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<meta name="msapplication-TileColor" content="#dfe3e7">
<meta name="theme-color" content="#dfe3e7">
<link rel="stylesheet"
      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
      integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
      crossorigin="anonymous">
<link rel="stylesheet"
      href="https://cdnjs.cloudflare.com/ajax/libs/open-iconic/1.1.1/font/css/open-iconic-bootstrap.min.css">
<title>Fairylab 路 Gameday</title>

<style>
html {
  min-height: 100%;
  overflow-y: scroll;
  position: relative;
}
header {
  background-color: #f5f5f5;
  width: 100%;
}
pre {
  margin: 0;
  white-space: pre-wrap;
  white-space: -moz-pre-wrap;
  white-space: -pre-wrap;
  white-space: -o-pre-wrap;
  word-wrap: break-word;
}
a.disabled{
  pointer-events: none;
  cursor: default;
}
.absolute-icon {
  position: absolute;
  top: 14px;
}
span.absolute-icon {
  border-radius: .25rem;
  padding: 3px;
  top: 12px;
}
.absolute-icon.left {
  left: 10px;
}
.absolute-icon.right {
  left: auto;
  right: 10px;
}
.absolute-icon.hover:hover {
  background-color: #e9ecef;
  cursor: pointer;
}
.alert, .dropdown-toggle, .form-control {
  padding: .75rem;
}
.alert .oi {
  line-height: 24px;
  padding-right: .75rem;
}
.anchor {
  padding-top: 0px;
  margin-top: 0px;
  transition: padding-top .35s ease, margin-top .35s ease;
}
.badge-icon {
  border: 1px solid #dee2e6;
  cursor: pointer;
  font-weight: 400!important;
  font-size: 100%;
  padding: .25rem .5rem;
  white-space: normal;
  width: 100%;
}
.badge-icon[data-toggle="modal"]:hover {
  background-color: #e9ecef;
  border-color: #dee2e6;
}
button.close {
  height: 16px;
  line-height: 16px;
  width: 16px;
}
.carousel-fade .carousel-item {
  opacity: 0;
  transition-duration: .6s;
  transition-property: opacity;
}
.carousel-fade .carousel-item.active,
.carousel-fade .carousel-item-next.carousel-item-left,
.carousel-fade .carousel-item-prev.carousel-item-right {
  opacity: 1;
}
.carousel-fade .active.carousel-item-left,
.carousel-fade .active.carousel-item-right {
  opacity: 0;
}
.carousel-fade .carousel-item-next,
.carousel-fade .carousel-item-prev,
.carousel-fade .carousel-item.active,
.carousel-fade .active.carousel-item-left,
.carousel-fade .active.carousel-item-prev {
  transform: translateX(0);
  transform: translate3d(0, 0, 0);
}
.dropdown-toggle {
 color: #6c767c;
}
.flex-grow {
  flex-grow: 1;
}
.grayscale {
  -webkit-filter: grayscale(1) opacity(0.5);
  filter: grayscale(1) opacity(0.5);
}
.head {
  border-left: 1px solid transparent;
  border-right: 1px solid transparent;
  padding: .75rem;
}
.h-72p {
  height: 72px;
}
.modal {
  overflow-y: auto;
}

.modal-open {
  overflow: auto;
  padding-right: 0px !important;
}

.mr-8p {
  margin-right: 8px;
}
.nav-pills {
  overflow-x: auto;
  overflow-y: hidden;
}
.navbar {
  padding: 0;
}
.nav-pills .nav-item {
  margin: 0 .25rem;
  min-width: 32px;
  text-align: center;
}
.nav-pills .nav-link {
  padding: .25rem 0;
}
.nav-pills .nav-link:not(.active) {
  border: 1px solid #dee2e6;
}
.page-link {
  background-color: #f8f9fa;
  line-height: 1;
  padding: .25rem .75rem;
}
.page-link-inner {
  color: #212529;
  display: inline-block;
  padding: 0;
  padding-top: 1px;
}
.pagination {
  margin: 0;
  position: absolute;
  right: 0;
  top: 4px;
}
.rt-10p {
  right: 10px;
  top: 10px;
}
.right-8p {
  right: 8px;
}
.table {
  margin-bottom: 0;
  word-break: break-word;
}
.table .thead-light td {
  background-color: #f5f5f5;
}
.table-fixed {
  table-layout: fixed;
}
.small.table td {
  padding: .75rem .5rem;
}
.small.table td:first-child {
  padding-left: .75rem;
}
.small.table td:last-child {
  padding-right: .75rem;
}
.small.table td.badge-icon-wrapper {
  padding: .5rem .25rem;
}
.small.topper {
  font-size: 90%;
}
.table.collapse.show {
  display: table;
}
.text-underline {
  text-decoration: underline;
}
.w-10 {
  width: 10%;
}
.w-20 {
  width: 20%;
}
.w-40 {
  width: 40%;
}
.w-24p {
  width: 24px;
}
.w-28p {
  width: 28px;
}
.w-32p {
  width: 32px;
}
.w-40p {
  width: 40px;
}
.w-50p {
  width: 50px;
}
.w-65p {
  width: 65px;
}
.w-75p {
  width: 75px;
}
.w-105p {
  width: 105px;
}
.w-150p {
  width: 150px;
}
.badge-icon-image,
.badge-icon-text {
  display: block;
}
.badge-icon-image {
  margin: 1px auto 0;
}
.badge-icon-text {
  color: #212529;
  padding: 0;
  padding-top: .25rem;
}
.td-sm-none,
.td-md-none,
.td-lg-none {
  display: none;
}
@media (min-width: 576px) {
  body.modal-open header {
    margin-right: 0!important;
    padding-right: 0!important;
  }
  .badge-icon-image,
  .badge-icon-text {
    display: inline-block;
  }
  .badge-icon-text {
    padding: 2px 0 2px .5rem;
  }
  .td-sm-none {
    display: table-cell;
  }
}
@media (min-width: 768px) {
  .badge-icon-image,
  .badge-icon-text {
    display: block;
  }
  .badge-icon-text {
    padding: 0;
    padding-top: .25rem;
  }
  .modal-dialog {
    max-width: 650px; 
  }
  .td-md-none {
    display: table-cell;
  }
}
@media (min-width: 992px) {
  .badge-icon-image,
  .badge-icon-text {
    display: inline-block;
  }
  .badge-icon-text {
    padding: 2px 0 2px .5rem;
  }
  .modal-dialog {
    max-width: 800px; 
  }
  .td-lg-none {
    display: table-cell;
  }
}
</style>
</head>

<body>

<!-- ^(base.header) -->
<header class="border-bottom sticky-top">
<div class="container">
<div class="head position-relative small text-secondary">
<img src="https://fairylab.surge.sh/favicon-32x32.png"
     width="16" height="16" border="0" class="absolute-icon left">
<div class="px-4">Gameday &nbsp;|&nbsp; 06:02:30 PDT (1985-10-26)</div>
<span class="oi oi-menu absolute-icon hover right"
      data-toggle="modal" data-target="#menu"></span></div>
</div>
</header>
<!-- $(base.header) -->

<!-- ^(base.main) -->
<main role="main" class="pt-3">

<div class="container">
<div class="row">
<div class="col-12">

<!-- ^(gameday.carousel) -->
<div id="gamedayCarousel" class="carousel slide carousel-fade" data-interval="false">

<!-- ^(gameday.carousel.inner) -->
<div class="carousel-inner">

<!-- ^(gameday.carousel.inner.days) -->
<div class="carousel-item active">

<!-- ^(gameday.carousel.inner.days.tables) -->
<!-- ^(common.table) -->
<table class="small table topper">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">Wednesday, August 28th, 2024</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold">Warmup</td>
<td scope="col" class="position-relative"><span class="oi oi-eye absolute-icon hover right text-dark" data-target="#game2449" data-toggle="modal"></span></td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative" colspan="2"><img src="https://fairylab.surge.sh/images/teams/tigers/tigers-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">Detroit</span></td>
</tr>
<tr>
<td scope="row" class="position-relative" colspan="2"><img src="https://fairylab.surge.sh/images/teams/twins/twins-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">Minnesota</span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border border-top-0 mb-3">

<!-- ^(common.table.foot) -->
<tfoot class="thead-light">
<tr>
<td scope="col" class="border-0 py-2"><span class="small text-secondary">7:10 PM at Target Field</span></td>
</tr>
<tr>
<td scope="col"><a href="/gameday/2449/">Gameday</a></td>
</tr>
</tfoot>
<!-- $(common.table.foot) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Pending</td>
<td scope="col" class="position-relative"><span class="oi oi-eye absolute-icon hover right text-dark" data-target="#date0828" data-toggle="modal"></span></td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row"  colspan="2">Arizona 路 Los Angeles</td>
</tr>
<tr>
<td scope="row"  colspan="2">Los Angeles 路 New York</td>
</tr>
<tr>
<td scope="row"  colspan="2">New York 路 San Francisco</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- $(gameday.carousel.inner.days.tables) -->

</div>
<div class="carousel-item">

<!-- ^(gameday.carousel.inner.days.tables) -->
<!-- ^(common.table) -->
<table class="small table topper">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">Thursday, August 29th, 2024</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold">Warmup</td>
<td scope="col" class="position-relative"><span class="oi oi-eye absolute-icon hover right text-dark" data-target="#game2469" data-toggle="modal"></span></td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative" colspan="2"><img src="https://fairylab.surge.sh/images/teams/tigers/tigers-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">Detroit</span></td>
</tr>
<tr>
<td scope="row" class="position-relative" colspan="2"><img src="https://fairylab.surge.sh/images/teams/twins/twins-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">Minnesota</span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border border-top-0 mb-3">

<!-- ^(common.table.foot) -->
<tfoot class="thead-light">
<tr>
<td scope="col" class="border-0 py-2"><span class="small text-secondary">7:10 PM at Target Field</span></td>
</tr>
<tr>
<td scope="col"><a href="/gameday/2469/">Gameday</a></td>
</tr>
</tfoot>
<!-- $(common.table.foot) -->

</table>
<!-- $(common.table) -->

<!-- $(gameday.carousel.inner.days.tables) -->

</div>
<div class="carousel-item">

<!-- ^(gameday.carousel.inner.days.tables) -->
<!-- ^(common.table) -->
<table class="small table topper">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">Friday, August 30th, 2024</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold">Warmup</td>
<td scope="col" class="position-relative"><span class="oi oi-eye absolute-icon hover right text-dark" data-target="#game2476" data-toggle="modal"></span></td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative" colspan="2"><img src="https://fairylab.surge.sh/images/teams/tigers/tigers-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">Detroit</span></td>
</tr>
<tr>
<td scope="row" class="position-relative" colspan="2"><img src="https://fairylab.surge.sh/images/teams/twins/twins-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">Minnesota</span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border border-top-0 mb-3">

<!-- ^(common.table.foot) -->
<tfoot class="thead-light">
<tr>
<td scope="col" class="border-0 py-2"><span class="small text-secondary">7:10 PM at Target Field</span></td>
</tr>
<tr>
<td scope="col"><a href="/gameday/2476/">Gameday</a></td>
</tr>
</tfoot>
<!-- $(common.table.foot) -->

</table>
<!-- $(common.table) -->

<!-- $(gameday.carousel.inner.days.tables) -->

</div>
<!-- $(gameday.carousel.inner.days) -->

</div>
<!-- $(gameday.carousel.inner) -->

<!-- ^(gameday.carousel.nav) -->
<ul class="pagination">

<li class="page-item">
<a class="page-link" href="#gamedayCarousel" role="button" data-slide="prev">
<span class="page-link-inner">芦</span>
</a>
</li>

<li class="page-item">
<a class="page-link" href="#gamedayCarousel" role="button" data-slide="next">
<span class="page-link-inner">禄</span>
</a>
</li>

</ul>
<!-- $(gameday.carousel.nav) -->

</div>
<!-- $(gameday.carousel) -->

</div>
</div>
</div>

<!-- ^(common.dialog) -->
<div class="modal fade" id="menu" tabindex="-1" role="dialog"
     aria-labelledby="menu-label" aria-hidden="true">
<div class="modal-dialog" role="document">
<div class="modal-content">

<!-- ^(common.dialog.header) -->
<div class="modal-header d-block py-0">
<div class="head position-relative small">
<img src="https://fairylab.surge.sh/favicon-32x32.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block px-4">Fairylab</span>
<button type="button" class="close absolute-icon right m-0 p-0"
        data-dismiss="modal" aria-label="Close">
<span aria-hidden="true">&times;</span>
</button>
</div>
</div>
<!-- $(common.dialog.header) -->

<!-- ^(common.dialog.body) -->
<div class="modal-body pb-0">
<!-- ^(common.table) -->
<table class="small table topper">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">Site Links</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Tasks</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-timer absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/gameday/">Gameday</a></span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-people absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/news/">News</a></span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-spreadsheet absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/standings/">Standings</a></span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Other</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-dashboard absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/dashboard/">Dashboard</a></span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-home absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/">Home</a></span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(common.dialog.body) -->

</div>
</div>
</div>
<!-- $(common.dialog) --></main>
<!-- $(base.main) -->

<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
        integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN"
        crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
        integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q"
        crossorigin="anonymous"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/js/bootstrap.min.js"
        integrity="sha384-a5N7Y/aK3qNeh15eJKGWxsqtnX/wWdSZSKp+81YjTmS15nvnvxKHuzaWwXHDli+4"
        crossorigin="anonymous"></script>

</body>

</html>